### PR TITLE
v2.x - BUILD: add configury logic to specify patch dir as --with-platform-patches-dir=DIR

### PR DESCRIPTION
**Pull request by mike-dubman**

### CODE REVIEW

**Comment by mike-dubman** [comment]
:bot:assign: @hppritcha 
:bot:label:enhancement
:bot:milestone:v2.0.0

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/835/ for details.

File: config/opal_load_platform.m4 [hunk 1]
```diff
@@ -97,7 +101,12 @@ AC_DEFUN([OPAL_LOAD_PLATFORM], [
             AC_SUBST(OPAL_PARAM_FROM_PLATFORM, "no")
         fi
 
-        patch_dir="${with_platform}.patches"
+        if test -d "$with_platform_patches_dir"; then
```
Comment (jsquyres) [inline config/opal_load_platform.m4]: I think you want to modify this test a bit.  If --with-platform-patches-dir=DIR is specified, and DIR does not exist, this should be an abort-configure kind of error (i.e., a human has asked for something that configure cannot deliver, therefore we should abort configure and left a human figure it out).

**Comment by mike-dubman** [comment]
@jsquyres - good catch, thanks - will fix that.

**Comment by hppritcha** [comment]
@jsquyres @miked-mellanox  is this ready to merge?

**Comment by jsquyres** [comment]
@hppritcha Doesn't look like @miked-mellanox has fixed the issue yet.  This was only submitted yesterday, after all.  :-)

**Comment by mike-dubman** [comment]
@jsquyres - please review and will propagate to master if OK.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/839/ for details.

File: config/opal_load_platform.m4 [hunk 2]
```diff
@@ -155,13 +172,15 @@ AC_DEFUN([OPAL_LOAD_PLATFORM], [
...
+                else
+                    AC_MSG_NOTICE([No platform patches in ${patch_dir}])
                 fi
 
             else
                 AC_MSG_WARN([Platform patches already applied, skipping. ${patch_done} can be removed to re-apply ])
             fi
-        else
-            AC_MSG_NOTICE([No platform patches in ${patch_dir}])
+          else
```
Comment (jsquyres) [inline config/opal_load_platform.m4]: Super minor nit: I think the indenting of "else" was already correct; now it's incorrect.

File: config/opal_load_platform.m4 [hunk 3]
```diff
@@ -155,13 +172,15 @@ AC_DEFUN([OPAL_LOAD_PLATFORM], [
...
+            AC_MSG_NOTICE([No platform patches ${patch_dir}])
```
Comment (jsquyres) [inline config/opal_load_platform.m4]: Minor nit: I think the "in" was correct ("No platform patches in (dir)").

**Comment by jsquyres** [comment]
Very minor nits, but otherwise ok.

File: config/opal_load_platform.m4 (hunk 3 above)
Comment (mike-dubman) [inline config/opal_load_platform.m4]: here patch_dir can be empty (line: 100), did not want to add "if"
https://github.com/miked-mellanox/ompi-release/commit/d1fe82ed6179ea894a556a0fe050ed324ce006e3#diff-d062f072b6901db0f80c53d37e0c80ceR110

**Comment by hppritcha** [comment]
Is this ready to merge?

**Comment by jsquyres** [comment]
@miked-mellanox Can you fix the whitespace / indenting problem? Thanks.

**Comment by mike-dubman** [comment]
@jsquyres - plz review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/877/ for details.

**Comment by jsquyres** [comment]
:+1:

**Comment by jsquyres** [comment]
@hppritcha Good to go

**Pull request merged**